.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,180 @@
+        with:
+          path: src
+          ref: ${{ env.TARGET_REF }}
+
+      - name: Install depot tools
+        # NOTE: can't use actions/checkout here because this is not hosted on
+        # GitHub.
+        run: |
+          git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+          echo "${GITHUB_WORKSPACE}/depot_tools" >> $GITHUB_PATH
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 2.7
+
+      - name: Install macOS SDK 10.3 (macOS only)
+        if: matrix.os_name == 'osx'
+        # NOTE: macOS 11 doesn't work with our (old) version of Chromium build,
+        # and the latest Chromium build doesn't work with Packager's build
+        # system.  To work around this, we need an older SDK version, and to
+        # get that, we need an older XCode version.  XCode 10.3 has SDK 10.14,
+        # which works.
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: 10.3
+
+      - name: Configure gclient
+        run: |
+          depot_tools/gclient config https://github.com/google/shaka-packager.git --name=src --unmanaged
+
+      - name: Sync gclient
+        run: |
+          depot_tools/gclient sync
+
+      - name: Build
+        run: |
+          depot_tools/ninja -C src/out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}
+
+      - name: Test
+        shell: bash
+        run: |
+          # NOTE: Some of these tests must be run from the "src" directory.
+          cd src
+          if [[ '${{ matrix.os_name }}' == 'osx' ]]; then
+            export DYLD_FALLBACK_LIBRARY_PATH=out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}
+          fi
+          for i in out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}/*test${{ matrix.exe_ext }}; do "$i" || exit 1; done
+          python out/${{ matrix.build_type }}${{ matrix.build_type_suffix }}/packager_test.py -v --libpackager_type=${{ matrix.lib_type }}_library
+
+      - name: Prepare artifacts
+        if: matrix.build_type == 'Release' && matrix.lib_type == 'static'
+        shell: bash
+        run: |
+          mkdir artifacts
+          mv \
+            src/out/Release${{ matrix.build_type_suffix }}/packager${{ matrix.exe_ext }} \
+            artifacts/packager-${{ matrix.os_name }}${{ matrix.exe_ext }}
+          mv \
+            src/out/Release${{ matrix.build_type_suffix }}/mpd_generator${{ matrix.exe_ext }} \
+            artifacts/mpd_generator-${{ matrix.os_name }}${{ matrix.exe_ext }}
+          if [[ '${{ matrix.os_name }}' == 'win' ]]; then
+            (
+              cd src/out/Release${{ matrix.build_type_suffix }}
+              7z a ../../../pssh-box.py.zip pyproto pssh-box.py
+            )
+            mv pssh-box.py.zip artifacts/pssh-box-${{ matrix.os_name }}.py.zip
+          else
+            tar -czf pssh-box.py.tar.gz -C src/out/Release${{ matrix.build_type_suffix }} pyproto pssh-box.py
```

Comment:
Can't you just put the output in the correct location?  Or would that affect the output layout?

Reply:
You're right, I could.  And I guess I probably should.  I was initially porting the original travis configs, which did it this way.  I just never stopped to re-evaluate that.  I'll make that change and test it on my fork to make sure it's working.